src/de.rs

```diff
@@ -1451,13 +1454,32 @@ impl<'a> Deserializer<'a> {
                         expected: "a value",
                         found: token.1.describe(),
                     },
-                ))
+                ));
             }
             None => return Err(self.eof()),
         };
         Ok(value)
     }
 
+    fn parse_keylike(&mut self, at: usize, span: Span, key: &'a str) -> Result<Value<'a>, Error> {
+        self.number_or_date(span, key).map_err(|_| {
```

Comment:
Oh sorry I don't mean to say that *all* errors should be translated to unquoted strings here, because some could legitimately be malformed numbers or something like that. I think it's probably just helpful to append/augment the existing error to indicate "you may have meant to use a string here"

Reply:
Is the new implementation closer to what you had in mind?